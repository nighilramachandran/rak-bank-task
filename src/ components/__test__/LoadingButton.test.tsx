import { render, screen, fireEvent } from "@testing-library/react";
import LoadingButton from "../loading-button";

describe("LoadingButton component", () => {
  test("renders button with children", () => {
    render(
      <LoadingButton onClick={() => {}} loading={false}>
        Click me
      </LoadingButton>
    );
    expect(screen.getByText("Click me")).toBeInTheDocument();
  });

  test("renders disabled button when loading is true", () => {
    render(
      <LoadingButton onClick={() => {}} loading={true}>
        Click me
      </LoadingButton>
    );
    const button = screen.getByRole("button");
    expect(button).toBeDisabled();
  });

  test("renders enabled button when loading is false", () => {
    render(
      <LoadingButton onClick={() => {}} loading={false}>
        Click me
      </LoadingButton>
    );
    const button = screen.getByRole("button");
    expect(button).not.toBeDisabled();
  });

  test("calls onClick handler when button is clicked", () => {
    const onClickMock = jest.fn();
    render(
      <LoadingButton onClick={onClickMock} loading={false}>
        Click me
      </LoadingButton>
    );
    const button = screen.getByRole("button");
    fireEvent.click(button);
    expect(onClickMock).toHaveBeenCalled();
  });

  test("renders CircularProgress when loading is true", () => {
    render(
      <LoadingButton onClick={() => {}} loading={true}>
        Click me
      </LoadingButton>
    );
    expect(screen.getByRole("progressbar")).toBeInTheDocument();
  });

  test("does not render CircularProgress when loading is false", () => {
    render(
      <LoadingButton onClick={() => {}} loading={false}>
        Click me
      </LoadingButton>
    );
    expect(screen.queryByRole("progressbar")).toBeNull();
  });
  test("match the snapshot", () => {
    const container = document.body;
    expect(container).toMatchSnapshot();
  });
});
