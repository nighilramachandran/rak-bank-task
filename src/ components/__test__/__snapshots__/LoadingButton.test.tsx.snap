// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingButton component match the snapshot 1`] = `<body />`;
